### PR TITLE
Inject request trace headers

### PR DESCRIPTION
**Pull request by stroem**

### CODE REVIEW

File: users/user_permission.go [hunk 1]
```diff
@@ -1,96 +1,74 @@
...
-
-	req.Header.Set("Authorization", identityToken)
-	req.Header.Set("Content-Type", "application/json")
+	req := client.Get("/users/{id}").
+		Assign("id", userID).
+		SetHeader(headers.ContentType, "application/json")
 
-	client := &http.Client{}
-	resp, err := client.Do(req)
+	rest := httpClientAccessMgmt(stage, identityToken)
```
Comment (lonnblad) [inline users/user_permission.go:36]: why "rest" and not "client" or "restClient"?

Comment (stroem) [reply users/user_permission.go:36]: Would be nice if you could explain why you don't like the naming "rest".
I will change it to "restClient"

Comment (lonnblad) [reply users/user_permission.go]: "rest" doesn't mean client and this is a client, the name should reflect what it is/what it's used for

File: hierarchy/hierarchy.go [hunk 2]
```diff
@@ -33,77 +51,49 @@ func Create(identityToken, stage, parentNodeID, label, description, nodetype, su
...
+		Assign("id", nodeID).
+		SetHeader(headers.ContentType, "application/json")
 
-	client := &http.Client{}
-	resp, err := client.Do(req)
+	restClient := httpClient(stage, identityToken)
+	resp, err := restClient.Do(ctx, req)
 	if err != nil {
-		return errors.Wrap(err, "client.Do failed")
+		return errors.Wrap(err, "failed to execute request")
```
Comment (johdah) [inline hierarchy/hierarchy.go:92]: Maybe switch to fmt.Errorf when you are changing stuff

**Review by johdah — APPROVED** [review]
Looks good, maybe change to fmt.Errorf when you are chaing stuff everywhere. But that's just a suggestion.

File: hierarchy/hierarchy.go (hunk 2 above)
Comment (stroem) [reply hierarchy/hierarchy.go:92]: Thats would be a different feature. Change all or nothing, not just a subset.

**Pull request merged**